docs/application/native/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1137 @@
+              /* Error handling */
+
+          if (0 >= count) {
+              /* Error handling */
+
+              return;
+          }
+
+          ret = vc_cmd_list_first(vc_cmd_list);
+          if (VC_ERROR_NONE != ret)
+              /* Error handling */
+
+          do {
+              ret = vc_cmd_list_get_current(vc_cmd_list, &vc_cmd);
+              if (VC_ERROR_NONE == ret) {
+                  ret = vc_cmd_get_command(vc_cmd, &command);
+                  if (VC_ERROR_NONE != ret)
+                      /* Error handling */
+
+                  if (NULL != command) {
+                      /* Command must be released */
+                      free(command);
+                  }
+
+                  ret = vc_cmd_get_type(vc_cmd, &type);
+                  if (VC_ERROR_NONE != ret)
+                      /* Error handling */
+                  /* Do not release the command handle */
+              }
+
+              ret = vc_cmd_list_next(vc_cmd_list);
+          } while (VC_ERROR_ITERATION_END != ret);
+      }
+      ```
+
+4.  Register the commands for recognition by setting the command list to the voice control. If you want to update the registered commands, set the command list again with the updated commands:
+
+    ```c
+    void
+    set_command(vc_cmd_list_h vc_cmd_list, int type)
+    {
+        int ret;
+        ret = vc_set_command_list(vc_cmd_list, type);
+        if (VC_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+5.  When no longer needed, unset the command list, and destroy the command and command list handles. When you destroy the command list, all the commands in the command list are not automatically released. If you no longer need the commands, set the second parameter of `vc_cmd_list_destroy()` to `true`:
+
+    ```c
+    /* Unset the command list */
+    void
+    unset_command(int type)
+    {
+        int ret;
+        ret = vc_unset_command_list(type);
+        if (VC_ERROR_NONE != ret)
+            /* Error handling */
+    }
+
+    /* Destroy the command handle */
+    void
+    destroy_command(vc_cmd_h vc_cmd)
+    {
+        int ret;
+        ret = vc_cmd_destroy(vc_cmd);
+        if (VC_ERROR_NONE != ret)
+            /* Error handling */
+    }
+
+    /* Destroy the command list handle */
+    void
+    destroy_command_list(vc_cmd_list_h vc_cmd_list)
+    {
+        int ret;
+        ret = vc_cmd_list_destroy(vc_cmd_list, true); /* true destroys all commands */
+        if (VC_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+<a name="register_command"></a>
+## Register commands
+
+- You can register the commands for recognition by setting the command list to the voice control service from a file that includes the commands. The parameter `file_path` is used to get the path of the file.
+  If you want to update the registered commands, set the command list again with the updated commands:
+
+  ```c
+  void set_command_list_from_file(const char* file_path)
```

Comment:
can we write like this ?
```suggestion
  void set_command_list_from_file(const char* <file_path>)

Reply:
This is kind of example code and your suggestion is wrong on C language grammar.
So, I am going to keep this code.